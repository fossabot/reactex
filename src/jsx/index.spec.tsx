import React from "react"
import {reactex, jsxtex } from '.'
import { renderToLaTeXString } from "../"

const toJSON = (o: any) => JSON.stringify(o, null, 2)


test('Simple JSX test', () => {
  expect(toJSON(jsxtex `<reactex_document></reactex_document>`)).toEqual(toJSON(<reactex_document></reactex_document>))
})


test('Simple JSX test', () => {
  expect(toJSON(reactex('<reactex_document></reactex_document>'))).toEqual(toJSON(<reactex_document></reactex_document>))
})


test('More complex JSX test', () => {
  expect(toJSON(reactex(`<reactex_document>The square of five is written as <reactex_math display key="tmp">5^2</reactex_math></reactex_document>`)))
  .toEqual(toJSON(<reactex_document>The square of five is written as <reactex_math display key="tmp">5^2</reactex_math></reactex_document>))
})

test('More complex JSX test: newline', () => {
  expect(toJSON(reactex(`<reactex_document>The square of five 
  is written as <reactex_math display key="tmp">5^2</reactex_math></reactex_document>`)))
  .toEqual(toJSON(<reactex_document>The square of five
     is written as <reactex_math display key="tmp">5^2</reactex_math></reactex_document>))
})


test('Even more complex JSX test', () => {
  const o1 = toJSON(jsxtex `
  <reactex_document>
    <reactex_new_command name="tmpArg" content="" />
    <reactex_new_environment name="centerOther" optionalArgs={['endText']} nbMandatoryArgs={1} start="\renewcommand{\tmpArg}{#1}\begin{center}#2\\" end="\\\tmpArg\end{center}"/>
    <reactex_environment name="centerOther" args={[{optional: false, value: 'before'}]}>
      This should be centered with a text before and default one after
    </reactex_environment>
    <reactex_renew_environment name="centerOther" optionalArgs={['other endText']} nbMandatoryArgs={1} start="\renewcommand{\tmpArg}{#1}\begin{center}#2\\" end="\\\tmpArg\end{center}"/>
    <reactex_environment name="centerOther" args={[{optional: false, value: 'before'}]}>
      This should be centered with a text before and after
    </reactex_environment>
  </reactex_document>`)

  const o2 = toJSON(<reactex_document>
    <reactex_new_command name="tmpArg" content="" />
    <reactex_new_environment name="centerOther" optionalArgs={['endText']} nbMandatoryArgs={1} start="\renewcommand{\tmpArg}{#1}\begin{center}#2\\" end="\\\tmpArg\end{center}"/>
    <reactex_environment name="centerOther" args={[{optional: false, value: 'before'}]}>
      This should be centered with a text before and default one after
    </reactex_environment>
    <reactex_renew_environment name="centerOther" optionalArgs={['other endText']} nbMandatoryArgs={1} start="\renewcommand{\tmpArg}{#1}\begin{center}#2\\" end="\\\tmpArg\end{center}"/>
    <reactex_environment name="centerOther" args={[{optional: false, value: 'before'}]}>
      This should be centered with a text before and after
    </reactex_environment>
  </reactex_document>)

  expect(o1).toEqual(o2)
})

test('Even more complex JSX test', () => {
  const o1 = toJSON(reactex(String.raw`
  <reactex_document>
    <reactex_new_command name="tmpArg" content="" />
    <reactex_new_environment name="centerOther" optionalArgs={['endText']} nbMandatoryArgs={1} start="\renewcommand{\tmpArg}{#1}\begin{center}#2\\" end="\\\tmpArg\end{center}"/>
    <reactex_environment name="centerOther" args={[{optional: false, value: 'before'}]}>
      This should be centered with a text before and default one after
    </reactex_environment>
    <reactex_renew_environment name="centerOther" optionalArgs={['other endText']} nbMandatoryArgs={1} start="\renewcommand{\tmpArg}{#1}\begin{center}#2\\" end="\\\tmpArg\end{center}"/>
    <reactex_environment name="centerOther" args={[{optional: false, value: 'before'}]}>
      This should be centered with a text before and after
    </reactex_environment>
  </reactex_document>`))

  const o2 = toJSON(<reactex_document>
    <reactex_new_command name="tmpArg" content="" />
    <reactex_new_environment name="centerOther" optionalArgs={['endText']} nbMandatoryArgs={1} start="\renewcommand{\tmpArg}{#1}\begin{center}#2\\" end="\\\tmpArg\end{center}"/>
    <reactex_environment name="centerOther" args={[{optional: false, value: 'before'}]}>
      This should be centered with a text before and default one after
    </reactex_environment>
    <reactex_renew_environment name="centerOther" optionalArgs={['other endText']} nbMandatoryArgs={1} start="\renewcommand{\tmpArg}{#1}\begin{center}#2\\" end="\\\tmpArg\end{center}"/>
    <reactex_environment name="centerOther" args={[{optional: false, value: 'before'}]}>
      This should be centered with a text before and after
    </reactex_environment>
  </reactex_document>)

  expect(o1).toEqual(o2)
})


test('New Environment + renew Environment', () => {
  expect(renderToLaTeXString(jsxtex `
    <reactex_document>
      <reactex_new_command name="tmpArg" content="" />
      <reactex_new_environment name="centerOther" optionalArgs={['endText']} nbMandatoryArgs={1} start="\renewcommand{\tmpArg}{#1}\begin{center}#2\\" end="\\\tmpArg\end{center}"/>
      <reactex_environment name="centerOther" args={[{optional: false, value: 'before'}]}>
        This should be centered with a text before and default one after
      </reactex_environment>
      <reactex_renew_environment name="centerOther" optionalArgs={['other endText']} nbMandatoryArgs={1} start="\renewcommand{\tmpArg}{#1}\begin{center}#2\\" end="\\\tmpArg\end{center}"/>
      <reactex_environment name="centerOther" args={[{optional: false, value: 'before'}]}>
        This should be centered with a text before and after
      </reactex_environment>
    </reactex_document>`)).toMatchSnapshot()
})