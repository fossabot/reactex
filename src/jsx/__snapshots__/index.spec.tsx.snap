// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Environment + renew Environment 1`] = `
"\\\\begin{document}
\\\\newcommand{\\\\tmpArg}[0]{}
\\\\newenvironment{centerOther}[2][endText]
{\\\\renewcommand{\\\\tmpArg}{#1}\\\\begin{center}#2\\\\\\\\}
{\\\\\\\\\\\\tmpArg\\\\end{center}}
\\\\begin{centerOther}{before}
This should be centered with a text before and default one after
\\\\end{centerOther}
\\\\renewenvironment{centerOther}[2][other endText]
{\\\\renewcommand{\\\\tmpArg}{#1}\\\\begin{center}#2\\\\\\\\}
{\\\\\\\\\\\\tmpArg\\\\end{center}}
\\\\begin{centerOther}{before}
This should be centered with a text before and after
\\\\end{centerOther}
\\\\end{document}"
`;
